From d8eb4ac41d881a19bea7673d753ba92e6a11f5d6 Mon Sep 17 00:00:00 2001
From: Nikita Popov <npopov@redhat.com>
Date: Thu, 23 Jan 2025 14:35:32 +0100
Subject: [PATCH] [Support] Remove ciso646 include (#123578)

This header has been removed in C++20 and causes a large amount of
deprecation spam when building against libstdc++ 15 in C++17 mode.

As far as I understand, we just need to include *some* STL header to get
access to the version macros, and as this header also includes
<optional> nowadays we can just drop the <cstd646> include entirely.
---
 llvm/include/llvm/Support/Threading.h | 1 -
 1 file changed, 1 deletion(-)

Index: git/llvm/include/llvm/Support/Threading.h
===================================================================
--- git.orig/llvm/include/llvm/Support/Threading.h
+++ git/llvm/include/llvm/Support/Threading.h
@@ -18,7 +18,6 @@
 #include "llvm/ADT/StringRef.h"
 #include "llvm/Config/llvm-config.h" // for LLVM_ON_UNIX
 #include "llvm/Support/Compiler.h"
-#include <ciso646> // So we can check the C++ standard lib macros.
 
 #if defined(_MSC_VER)
 // MSVC's call_once implementation worked since VS 2015, which is the minimum
